hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java

```diff
@@ -536,24 +560,62 @@ public boolean scheduleExists(ScheduleID id) {
 
 		List<TokenNftInfo> nftInfos = new ArrayList<>();
 		var uniqueTokensMap = uniqueTokens.get();
-		uniqueTokenAccountOwnerships.get()
-				.get(fromGrpcAccountId(aid), (int) start, (int) end).forEachRemaining(nftId -> {
-			var nft = uniqueTokensMap.get(nftId);
-			nftInfos.add(
-					TokenNftInfo.newBuilder()
-							.setAccountID(aid)
-							.setCreationTime(nft.getCreationTime().toGrpc())
-							.setNftID(NftID.newBuilder()
-									.setTokenID(nftId.tokenId().toGrpcTokenId())
-									.setSerialNumber(nftId.serialNumber())
-									.build())
-							.setMetadata(ByteString.copyFrom(nft.getMetadata()))
-							.build()
-			);
-		});
+		var uniqueTokenAccountCount = uniqueTokenAccountOwnerships.get().getCount(fromGrpcAccountId(aid));
+
+		if (uniqueTokenAccountCount >= end) {
+			uniqueTokenAccountOwnerships.get()
+					.get(fromGrpcAccountId(aid), (int) start, (int) end).forEachRemaining(nftId -> {
+				var nft = uniqueTokensMap.get(nftId);
+				nftInfos.add(
+						buildTokenNft(aid, nft, nftId)
+				);
+			});
+		} else if (uniqueTokenAccountCount <= start) {
+			var tokens = tokenStore.listOfTokensServed(aid);
+			for (var tid : tokens) {
+				uniqueTokenTreasuryOwnerships.get()
+						.get(fromGrpcTokenId(tid), (int) start, (int) end).forEachRemaining(nftId -> {
```

Comment:
We shouldn't query the `start` here. If the `start = 50` and the `uniqueTokenAccountOwnerships` has `40`. The start should be `10` for the `uniqueTokenTreasuryOwnerships`